### PR TITLE
components unit test

### PR DESCRIPTION
**Pull request by leangseu-edx**
https://2u-internal.atlassian.net/browse/AU-873

### CODE REVIEW

File: src/components/Banner.test.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  describe('snapshot', () => {
+    it('renders default banner', () => {
+      const wrapper = shallow(<Banner {...props} />);
+      expect(wrapper).toMatchSnapshot();
+    });
+    it('renders with variants', () => {
+      const wrapper = shallow(<Banner {...props} variant="success" />);
+      expect(wrapper).toMatchSnapshot();
+
+      expect(wrapper.find('Alert').prop('variant')).toEqual('success');
```
Comment (muselesscreator) [inline src/components/Banner.test.jsx:18]: only nit is that I would prefer to find by imported component vs string.
`wrapper.find(Alert)` (which I *believe* works?)

I would PREFER not to have any magic strings here at all, but that would take some added key-stores for props and variants. (I do think that would make sense at some point, as its not a huge list we would want to track)

File: src/containers/CourseCard/components/CourseCardActions/BeginCourseButton.test.jsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+describe('BeginCourseButton', () => {
+  const props = {
+    cardId: 'cardId',
+  };
+  hooks.useCardCourseRunData.mockReturnValue({
+    homeUrl: 'homeUrl',
+  });
+  describe('snapshot', () => {
+    it('renders default button', () => {
```
Comment (muselesscreator) [inline src/containers/CourseCard/components/CourseCardActions/BeginCourseButton.test.jsx:21]:
```suggestion
    it('renders default button when learner has access to the course', () => {
```

File: src/containers/CourseCard/components/CourseCardActions/BeginCourseButton.test.jsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      hooks.useCardEnrollmentData.mockReturnValueOnce({
+        hasAccess: true,
+      });
+      const wrapper = shallow(<BeginCourseButton {...props} />);
+      expect(wrapper).toMatchSnapshot();
+    });
+    it('renders disabled button', () => {
```
Comment (muselesscreator) [inline src/containers/CourseCard/components/CourseCardActions/BeginCourseButton.test.jsx:28]:
```suggestion
    it('renders disabled button when learner does not have access to the course', () => {
```

File: src/containers/CourseCard/components/CourseCardActions/ResumeButton.test.jsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+describe('ResumeButton', () => {
+  const props = {
+    cardId: 'cardId',
+  };
+  hooks.useCardCourseRunData.mockReturnValue({
+    resumeUrl: 'resumeUrl',
+  });
+  describe('snapshot', () => {
+    it('renders default button', () => {
```
Comment (muselesscreator) [inline src/containers/CourseCard/components/CourseCardActions/ResumeButton.test.jsx:21]:
```suggestion
    it('renders default button when learner has access to the course', () => {
```

File: src/containers/CourseCard/components/CourseCardActions/ResumeButton.test.jsx [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      hooks.useCardEnrollmentData.mockReturnValueOnce({
+        hasAccess: true,
+      });
+      const wrapper = shallow(<ResumeButton {...props} />);
+      expect(wrapper).toMatchSnapshot();
+    });
+    it('renders disabled button', () => {
```
Comment (muselesscreator) [inline src/containers/CourseCard/components/CourseCardActions/ResumeButton.test.jsx:28]:
```suggestion
    it('renders disabled button when learner does not have access to the course', () => {
```

File: src/containers/CourseCard/components/CourseCardActions/ResumeButton.test.jsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      hooks.useCardEnrollmentData.mockReturnValueOnce({
+        hasAccess: false,
+      });
+      const wrapper = shallow(<ResumeButton {...props} />);
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+  describe('behavior', () => {
+    it('disabled button when audit access expired', () => {
```
Comment (muselesscreator) [inline src/containers/CourseCard/components/CourseCardActions/ResumeButton.test.jsx:37]:
```suggestion
    it('renders disabled button when audit access expired', () => {
```

File: src/containers/CourseCard/components/CourseCardActions/ResumeButton.test.jsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      hooks.useCardEnrollmentData.mockReturnValueOnce({
+        hasAccess: true,
+        isAudit: true,
+        isAuditAccessExpired: true,
+      });
+      const wrapper = shallow(<ResumeButton {...props} />);
+      expect(wrapper.prop('disabled')).toEqual(true);
+    });
+    it('enabled button when audit access not expired', () => {
```
Comment (muselesscreator) [inline src/containers/CourseCard/components/CourseCardActions/ResumeButton.test.jsx:46]:
```suggestion
    it('renders enabled button when audit access not expired', () => {
```

File: src/containers/CourseCard/components/CourseCardActions/SelectSessionButton.test.jsx [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+      hasSessions,
+    });
+    return shallow(<SelectSessionButton {...props} />);
+  };
+  describe('snapshot', () => {
+    it('renders default button', () => {
+      const wrapper = createWrapper({ hasAccess: true, canChange: true, hasSessions: true });
+      expect(wrapper).toMatchSnapshot();
+    });
+    it('renders disabled button', () => {
```
Comment (muselesscreator) [inline src/containers/CourseCard/components/CourseCardActions/SelectSessionButton.test.jsx:37]:
```suggestion
    it('renders disabled button when user does not have access to the course', () => {
```

File: src/containers/CourseCard/components/CourseCardActions/SelectSessionButton.test.jsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  });
+  describe('behavior', () => {
+    it('default render', () => {
+      const wrapper = createWrapper({ hasAccess: true, canChange: true, hasSessions: true });
+      expect(wrapper.prop('disabled')).toEqual(false);
+      expect(wrapper.prop('href')).toEqual('resumeUrl');
+      expect(wrapper.prop('onClick').getMockName())
+        .toEqual(hooks.useUpdateSelectSessionModalCallback().getMockName());
+    });
+    it('disabled iif doesn\'t have access or user cannot change or doesn\'t have sessions', () => {
```
Comment (muselesscreator) [inline src/containers/CourseCard/components/CourseCardActions/SelectSessionButton.test.jsx:50]:
```suggestion
    it('disabled if learner doesn\'t have access, cannot change sessions, or does not have sessions', () => {
```

File: src/containers/CourseCard/components/CourseCardActions/UpgradeButton.test.jsx [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+}));
+
+describe('UpgradeButton', () => {
+  const props = {
+    cardId: 'cardId',
+  };
+  const upgradeUrl = 'upgradeUrl';
+  hooks.useCardCourseRunData.mockReturnValue({ upgradeUrl });
+  describe('snapshot', () => {
+    it('can upgrade', () => {
```
Comment (muselesscreator) [inline src/containers/CourseCard/components/CourseCardActions/UpgradeButton.test.jsx:20]: nit: I prefer `test` for snapshot tests.
reads as "Snapshot - test: can upgrade"
vs "Snapshot - it: can upgrade".
`it` tests read better for behavior.

Comment (muselesscreator) [reply src/containers/CourseCard/components/CourseCardActions/UpgradeButton.test.jsx:20]: i.e. `it('renders')`, `it('provides exit action when <...>')`, etc

File: src/containers/CourseList/index.jsx [hunk 11]
```diff
@@ -1,48 +1,19 @@
...
 import { ActiveCourseFilters, CourseFilterControls } from 'containers/CourseFilterControls';
 import CourseCard from 'containers/CourseCard';
 
+import { useCourseListData } from './hooks';
+
 import messages from './messages';
 
 import './index.scss';
 
-export const useCourseListData = () => {
```
Comment (muselesscreator) [inline src/containers/CourseList/index.jsx:18]: ++

File: src/containers/CourseList/index.test.jsx [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+      const wrapper = createWrapper({
+        filterOptions: {
+          abitary: 'filter',
+        },
+        showFilters: true,
+      });
+      expect(wrapper).toMatchSnapshot();
+    });
+    test('with multiple courses', () => {
+      const wrapper = createWrapper({
```
Comment (muselesscreator) [inline src/containers/CourseList/index.test.jsx:47]: Request: can we get a multi-page snapshot (numPages > 1)?

File: src/containers/Dashboard/index.test.jsx [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+  });
+
+  describe('behavior', () => {
+    it('initializes the app without courses', () => {
+      const wrapper = createWrapper({
+        hasCourses: false,
+        hasAvailableDashboards: false,
+        showSelectSessionModal: false,
+      });
+      expect(wrapper.find('EmptyCourse').length).toEqual(1);
```
Comment (muselesscreator) [inline src/containers/Dashboard/index.test.jsx:83]: Similar request that if possible I would rather we find by the component definition than a magic string

File: src/containers/LearnerDashboardHeader/GreetingBanner.jsx [hunk 14]
```diff
@@ -3,12 +3,14 @@ import classNames from 'classnames';
...
 
 import { getConfig } from '@edx/frontend-platform';
-import { FormattedMessage } from '@edx/frontend-platform/i18n';
+import { useIntl } from '@edx/frontend-platform/i18n';
 import { Image } from '@edx/paragon';
 
 import messages from './messages';
 
 export const GreetingBanner = ({ size }) => {
+  const { formatMessage } = useIntl();
```
Comment (muselesscreator) [inline src/containers/LearnerDashboardHeader/GreetingBanner.jsx:12]: yeeeesss... kill AAAALLL the old syntax!

**Review by muselesscreator — APPROVED** [review]
Minor nits/requests, but looks great!

File: src/containers/CourseList/index.test.jsx (hunk 12 above)
Comment (leangseu-edx) [reply src/containers/CourseList/index.test.jsx:47]: Sure. IMHO, I don't think it would make much of a different.

**Pull request merged**